ambry-rest/src/main/java/com.github.ambry.rest/RestServer.java

```diff
@@ -157,8 +159,17 @@ public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterM
     RestRequestMetricsTracker.setDefaults(metricRegistry);
     restServerState = new RestServerState(restServerConfig.restServerHealthCheckUri);
     restServerMetrics = new RestServerMetrics(metricRegistry, restServerState);
+
+    boolean sslRequired = restServerConfig.restServerEnableHTTPS
+        || new ClusterMapConfig(verifiableProperties).clusterMapSslEnabledDatacenters.length() > 0;
+    SSLFactory sslFactory = null;
+    if (sslRequired) {
+      sslFactory = Utils.getObj(restServerConfig.restServerSSLFactory, verifiableProperties);
```

Comment:
ok this looks a bit iffy to me. Even though I set `restServerConfig.restServerEnableHTTPS` to `false`, I'm supposed to set a valid value for `restServerConfig.restServerSSLFactory` just because `Router` might need it. And `RestServer` knows that it can deduce whether that the `Router` might need it by looking at a cluster map config. Also most of the configs required for ssl are in `SSLConfig` which is yet another place to look for ssl init configs.

I have an alternate suggestion. I suggest dropping reflection for this (why do reflection anyway?). Pass the  `SSLFactory` (or `null`) from `AmbryRestServerMain` to `RestServer` based on arguments received (i.e. enable/disable SSL). There is precedent for this because `ClusterMap` is passed this way. Each of the components (`RestServer` and `Router`) can then make decisions on using `SSLFactory` based on their own decision making process. 

Also, I'm still unsure whether `enableHttps` should be a rest server config or netty server config (since the ports are netty configs).